include/bluetooth/mesh/gen_dtt_srv.h

```diff
@@ -107,20 +108,32 @@ void bt_mesh_dtt_srv_set(struct bt_mesh_dtt_srv *srv, uint32_t transition_time);
 int bt_mesh_dtt_srv_pub(struct bt_mesh_dtt_srv *srv,
 			struct bt_mesh_msg_ctx *ctx);
 
-/** @brief Find the Generic DTT server in a given element.
+/** @brief Find the Generic DTT server over multiple elements.
  *
- * @param[in] elem Element to find the DTT server in.
+ * If the DTT Server is not present on the given element of the model, then the
+ * try to find the DTT Server model instance that is present on the element with
+ * the largest address that is smaller than the address of the given element.
+ *
+ * @param[in] elem Element to start search of the DTT server in.
  *
  * @return A pointer to the DTT server instance, or NULL if no instance is
  * found.
  */
 static inline struct bt_mesh_dtt_srv *
 bt_mesh_dtt_srv_get(const struct bt_mesh_elem *elem)
 {
-	struct bt_mesh_model *model = bt_mesh_model_find(
-		elem, BT_MESH_MODEL_ID_GEN_DEF_TRANS_TIME_SRV);
-
-	return (struct bt_mesh_dtt_srv *)(model ? model->user_data : NULL);
+	struct bt_mesh_model *model = NULL;
```

Comment:
This is quite not optimal to have complex functions as inline. I'd recommend moving function body to *.c file and leave here only declaration.
```
struct bt_mesh_dtt_srv *bt_mesh_dtt_srv_get(const struct bt_mesh_elem *elem);
```

Reply:
That was my first thought too, but then we have to add this model to all the samples as they only uses this two inline functions.

Reply:
What about
```
#ifdef BT_MESH_DTT_SRV
struct bt_mesh_dtt_srv *
bt_mesh_dtt_srv_get(const struct bt_mesh_elem *elem);
#else
static inline struct bt_mesh_dtt_srv *
bt_mesh_dtt_srv_get(const struct bt_mesh_elem *elem)
{
        return NULL;
}
#endif
```
?

Reply:
Or you could `#ifdef` everywhere the function gets called, that's perhaps less "magic"

Reply:
It is better to create empty stub instead of linking never used functionality.